lib/Resizable.js

```diff
@@ -4,227 +4,492 @@ import PropTypes from 'prop-types';
-          key="resizableHandle"
-          onStop={this.resizeHandler('onResizeStop')}
-          onStart={this.resizeHandler('onResizeStart')}
-          onDrag={this.resizeHandler('onResize')}
-          >
-          <span className="react-resizable-handle" />
-        </DraggableCore>
-      ]
-    });
-  }
+
+
+    componentWillReceiveProps(nextProps: Object) {
+        if (this.state.resizing) {
+            return
+        }
+        if (nextProps.width !== this.props.width) {
+            this.setState({width: nextProps.width})
+        }
+        if (nextProps.height !== this.props.height) {
+            this.setState({height: nextProps.height})
+        }
+        if (nextProps.left !== this.props.left) {
+            this.setState({left: nextProps.left})
+        }
+        if (nextProps.top !== this.props.top) {
+            this.setState({top: nextProps.top})
+        }
+    }
+
+    lockAspectRatio(width: number, height: number, aspectRatio: number): [number, number] {
+        height = width / aspectRatio;
+        width = height * aspectRatio;
+        return [width, height];
+    }
+
+    runConstraints(width: number, height: number): [number, number] {
+        const [min, max] = [this.props.minConstraints, this.props.maxConstraints];
+
+        if (this.props.lockAspectRatio) {
+            const ratio = this.state.width / this.state.height;
+            height = width / ratio;
+            width = height * ratio;
+        }
+
+        if (!min && !max) return [width, height];
+
+        const [oldW, oldH] = [width, height];
+
+        // Add slack to the values used to calculate bound position. This will ensure that if
+        // we start removing slack, the element won't react to it right away until it's been
+        // completely removed.
+        let {slackW, slackH} = this.state;
+        width += slackW;
+        height += slackH;
+
+        if (min) {
+            width = Math.max(min[0], width);
+            height = Math.max(min[1], height);
+        }
+        if (max) {
+            width = Math.min(max[0], width);
+            height = Math.min(max[1], height);
+        }
+
+        // If the numbers changed, we must have introduced some slack. Record it for the next iteration.
+        slackW += (oldW - width);
+        slackH += (oldH - height);
+        if (slackW !== this.state.slackW || slackH !== this.state.slackH) {
+            this.setState({slackW, slackH});
+        }
+
+        return [width, height];
+    }
+
+    increaseHeightMoveUp = (currentTop, y, dimension, deltaY) => {
+        if (y < this.state.startY) {
+            let heightToAdd = Math.abs(deltaY)
+            if (this.props.maxConstraints[1] &&
+                heightToAdd + dimension.height > this.props.maxConstraints[1]) {
+                heightToAdd = this.props.maxConstraints[1] - dimension.height
+            }
+            dimension.height = this.state.height + heightToAdd
+            dimension.top = currentTop - heightToAdd
```

Comment:
It's not safe to mutate function input. Let's refactor this function in a way we don't change 'dimension' property. 
I propose this function returns {height, top} only which you can merge in parent function with width and left (resizeFromTopLeftCorner). 

Same for other functions

Reply:
done